lib/patch-https-request-for-proxying.js

```diff
@@ -80,11 +80,10 @@ function shouldProxy(uri) {
 // Entry point: To patch or not to patch?
 if (brokerServer.host && httpsProxy && shouldProxy(brokerServer)) {
   const proxy = url.parse(httpsProxy);
+  let tunnelProxy = ({ host: hostname, port } = proxy);
+  tunnelProxy = { ...tunnelProxy, ...(proxyAuth && { proxyAuth }) };
```

Comment:
```suggestion
  const {host: hostname, port} = url.parse(httpsProxy);
  const tunnelProxy = { ...{ host, port }, ...{proxyAuth} };
```
And make `proxyAuth` by default to return `{}`

Reply:
Shouldn't we just not add this property if it's not defined?

Reply:
fixed suggestion

Reply:
It's similar to what I wrote. It should behave exactly as you intended.

Reply:
I agree, but it's much more readable.

Reply:
I agree with @dkontorovskyy here, I find the suggested change more readable.